analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -262,6 +262,7 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
             var expectedIssue = expectedIssues
                 .Where(issueFilter)
                 .OrderBy(x => x.Start == null ? 0 : Math.Abs(location.GetLineSpan().StartLinePosition.Character - x.Start.Value))
+                .ThenBy(x => x.Message == message ? 0 : 1)  // Prefer issue with explicit matching message when possible. There can be more with different messages
```

Comment:
Bunch of tests started failing SymbolicExecutionRunnerTest`, because `ImmutableDictionary` doesn't sort the items the same way under `net48` and `net6`.
So single test required to be
```
            Verify(@"string s = null;   // Noncompliant    {{Message for SAll}}
                                        // Noncompliant@-1 {{Message for SMain}}
```
and
```
            Verify(@"string s = null;   // Noncompliant    {{Message for SMain}}
                                        // Noncompliant@-1 {{Message for SAll}}
```
at the same time.

This was annoying me for quite some time already because more than one compilation error had to be sorted in a specific order for no reason.